File: Challenge3/ClientApp/src/app/viewmodels/result.ts
export class Result {
  category: string;
  type: string;
  difficulty: string;
  question: string;
  correct_answer: string;
  incorrect_answers: string[];
  questWiseOptions: QuestWiseOption[];
  //mcq: string[];
  id: number;
  isSelected: boolean;
}

export class ResponseGroup {
  response_Code: string;
  results: Result[];
  //selectedAnswers: string[];
}

export class QuestWiseOption {
  option: string;
  selected: boolean;
}

export class SelectedAnswer {
  id: number;
  answer: string;
  isCorrect: boolean;
}

File: Challenge3/Models/Result.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Challenge3.Models
{
    public class Result
    {
        [JsonProperty("category")]
        public string Category { get; set; }
        [JsonProperty("type")]
        public string Type { get; set; }
        [JsonProperty("difficulty")]
        public string Difficulty { get; set; }
        [JsonProperty("question")]
        public string Question { get; set; }
        [JsonProperty("correct_answer")]
        public string Correct_answer { get; set; }
        [JsonProperty("incorrect_answers")]
        public List<string> Incorrect_answers { get; set; }


    }

    public class ApiResponse
    {
        public string Response_Code { get; set; }
        public List<Result> Results { get; set; }
    }
}
